Bottom: bafe3b7b2fed72d5abd50b4ae61c473900f4a3b7
Top:    bc86a460886b269af3c734d339098a5882be72cb
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-23 12:06:51 -0400

Temporary holding patch


---

diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 3e6838c..9f2df15 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -35,9 +35,9 @@ static unsigned short psc_ac97_read(struct snd_ac97 *ac97, unsigned short reg)
 	unsigned int val;
 
 	/* Wait for command send status zero = ready */
-	spin_event_timeout((in_be16(&psc_dma->psc_regs->sr_csr.status) &
+	spin_event_timeout(!(in_be16(&psc_dma->psc_regs->sr_csr.status) &
 				MPC52xx_PSC_SR_CMDSEND), 100, 0, rc);
-	if (rc != 0) {
+	if (rc == 0) {
 		pr_err("timeout on ac97 bus (rdy)\n");
 		return -ENODEV;
 	}
@@ -45,9 +45,9 @@ static unsigned short psc_ac97_read(struct snd_ac97 *ac97, unsigned short reg)
 	out_be32(&psc_dma->psc_regs->ac97_cmd, (1<<31) | ((reg & 0x7f) << 24));
 
 	/* Wait for the answer */
-	spin_event_timeout(!(in_be16(&psc_dma->psc_regs->sr_csr.status) &
+	spin_event_timeout((in_be16(&psc_dma->psc_regs->sr_csr.status) &
 				MPC52xx_PSC_SR_DATA_VAL), 100, 0, rc);
-	if (rc != 0) {
+	if (rc == 0) {
 		pr_err("timeout on ac97 read (val) %x\n",
 				in_be16(&psc_dma->psc_regs->sr_csr.status));
 		return -ENODEV;
@@ -69,9 +69,9 @@ static void psc_ac97_write(struct snd_ac97 *ac97,
 	int rc;
 
 	/* Wait for command status zero = ready */
-	spin_event_timeout((in_be16(&psc_dma->psc_regs->sr_csr.status) &
+	spin_event_timeout(!(in_be16(&psc_dma->psc_regs->sr_csr.status) &
 				MPC52xx_PSC_SR_CMDSEND), 100, 0, rc);
-	if (rc != 0) {
+	if (rc == 0) {
 		pr_err("timeout on ac97 bus (write)\n");
 		return;
 	}
@@ -86,7 +86,7 @@ static void psc_ac97_warm_reset(struct snd_ac97 *ac97)
 	struct mpc52xx_psc __iomem *regs = psc_dma->psc_regs;
 
 	out_be32(&regs->sicr, psc_dma->sicr | MPC52xx_PSC_SICR_AWR);
-	spin_event_timeout(1, 3, 0, rc);
+	spin_event_timeout(0, 3, 0, rc);
 	out_be32(&regs->sicr, psc_dma->sicr);
 }
 
@@ -97,9 +97,9 @@ static void psc_ac97_cold_reset(struct snd_ac97 *ac97)
 
 	/* Do a cold reset */
 	out_8(&regs->op1, MPC52xx_PSC_OP_RES);
-	spin_event_timeout(1, 10, 0, rc);
+	spin_event_timeout(0, 10, 0, rc);
 	out_8(&regs->op0, MPC52xx_PSC_OP_RES);
-	spin_event_timeout(1, 50, 0, rc);
+	spin_event_timeout(0, 50, 0, rc);
 	psc_ac97_warm_reset(ac97);
 }
 
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index bad9d88..cc1e618 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -20,6 +20,8 @@
  *   o Support TDM on PCM and I2S
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
@@ -125,7 +127,7 @@ static int soc_pcm_apply_symmetry(struct snd_pcm_substream *substream)
 
 	if (codec_dai->symmetric_rates || cpu_dai->symmetric_rates ||
 	    machine->symmetric_rates) {
-		dev_dbg(card->dev, "Symmetry forces %dHz rate\n", 
+		dev_dbg(card->dev, "Symmetry forces %dHz rate\n",
 			machine->rate);
 
 		ret = snd_pcm_hw_constraint_minmax(substream->runtime,
diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index c1222fa..f3ccbd0 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -29,6 +29,8 @@
  *    o Support for reduced codec bias currents.
  */
 
+#define DEBUG
+
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
